DiverseSelector/diversity.py

```diff
@@ -135,46 +133,18 @@ def entropy(x: np.ndarray) -> float:
     return e
 
 
-def explicit_diversity_index(
-    x: np.ndarray, mols: List[rdkit.Chem.rdchem.Mol],
-) -> float:
-    """Computes the explicit diversity index.
-
-    Parameters
-    ----------
-    x : ndarray
-        Feature matrix.
-    mols: List[rdkit.Chem.rdchem.Mol]
-        Molecules from feature matrix.
-
-    Returns
-    -------
-    edi_scaled : float
-        Explicit diversity index.
-
-    Notes
-    -----
-    This method hasn't been tested.
-    Papp, Á., Gulyás-Forró, A., Gulyás, Z., Dormán, G., Ürge, L.,
-    and Darvas, F.. (2006) Explicit Diversity Index (EDI):
-    A Novel Measure for Assessing the Diversity of Compound Databases.
-    Journal of Chemical Information and Modeling 46, 1898-1904.
-    """
-    cs = len(rdFMCS.FindMCS(mols))
-    nc = len(x)
-    sdi = (1 - nearest_average_tanimoto(x)) / (0.8047 - (0.065 * (np.log(nc))))
-    cr = -1 * np.log10(nc / (cs ** 2))
-    edi = (sdi + cr) * 0.7071067811865476
-    edi_scaled = ((np.tanh(edi / 3) + 1) / 2) * 100
-    return edi_scaled
```

Comment:
Is there any reason to get rid of this index, `explicit_diversity_index`?

Reply:
We wanted to remove RD Kit as a dependency, and as this function depended on it, we decided to remove it.

Reply:
@FanwangM, to keep this function, the API needs to change. The `mols` are only used to obtain `cs = len(rdFMCS.FindMCS(mols))`, so `cs` can be an argument. @AWBroscius can you give this a try?

Reply:
I can certainly go back and change the code to make `cs` an argument. However, I do not know what to write in the documentation for it. In the original paper, `cs` is the number of core structures/common-substructures present in a library of chemical compounds. How would this translate for non-chemical datasets?  Would it make sense to apply this measure to non-chemical datasets? @FarnazH @FanwangM

Reply:
Thanks for the clarification. I don't think it's applicable for non-chemical data. We should make this clear in the docstring and we are OK. Probably we can put this information into the `Notes` section. @awb9691